File: golang/structure/adaptor.go
package main

type Adaptee struct {

}
func (a *Adaptee) f1() {
	println("f1")
}

type Adaptor struct {
	adptee *Adaptee
}
func NewAdaptor(a *Adaptee) *Adaptor {
	return &Adaptor{a}
}
func (ar *Adaptor) fa() {
	ar.adptee.f1()
}
func main() {
	ar := NewAdaptor(&Adaptee{})
	ar.fa()
}

File: cpp/structure/adaptor.cpp
#include <iostream>
using namespace std;
class Adaptee {
public:
    void f1() {
        cout<< "f1" <<endl;
    }
};
class Adaptor: public Adaptee {
public:
    void fa() {
        f1();
    }
};
int main() {
    Adaptor adaptor;
    adaptor.fa();
}